maistra/run-ci.sh

```diff
@@ -23,53 +23,76 @@ bazel-bin/source/exe/envoy-static --version
 # The following build step helps reduce resources usage
 # by compiling tests first.
 # Build tests
+
 time bazel build \
   ${COMMON_FLAGS} \
   --build_tests_only \
   -- \
   //test/... \
   -//test/extensions/listener_managers/listener_manager:listener_manager_impl_quic_only_test
 
-# These tests can fail with TIME OUT if run as part of batch
-extra_tests=(
-    //test/exe:main_common_test \
-    //test/server:guarddog_impl_test \
-    //test/common/stream_info:filter_state_impl_test \
-    //test/common/common:assert_enabled_in_release_test \
-    //test/common/common:assert_test \
-    //test/common/upstream:conn_pool_map_impl_test \
-    //test/common/conn_pool:conn_pool_base_test \
-    //test/common/http/http2:codec_impl_test \
-    //test/extensions/filters/http/header_mutation:header_mutation_test \
-    //test/extensions/filters/http/gcp_authn:gcp_authn_filter_test \
-    //test/extensions/filters/http/ext_proc:filter_test \
-    //test/extensions/filters/http/custom_response:config_test \
-    //test/extensions/filters/http/ext_proc:streaming_integration_test \
-    //test/extensions/filters/network/http_connection_manager:config_test )
-
-EXCLUDE_BATCH_TESTS=""
-for test in "${extra_tests[@]}"
-do
-  EXCLUDE_BATCH_TESTS+=" -${test}"
-done
-
+function run_tests() {
 # Run tests without tests that timeout
 time bazel test \
   ${COMMON_FLAGS} \
   --build_tests_only \
+  --spawn_strategy=local \
   -- \
-  //test/... \
-  -//test/extensions/listener_managers/listener_manager:listener_manager_impl_quic_only_test \
-  $EXCLUDE_BATCH_TESTS
+  $1
+}
 
-# Run tests that where timing out in batch
-for test in "${extra_tests[@]}"
-do
-echo "Running Test " $test
-time bazel test \
-  ${COMMON_FLAGS} \
-  --build_tests_only \
-  -- \
-  $test 
-done
+# --cache_test_results=no \
+
+# Run the tests in smaller blocks rather than attempt to 
+# share resources to run //test/... 
+# This is a remedy for the tests that would time out.
+run_tests "//test/common/..."
```

Comment:
Rather than hard coding their names, it would be better to have a for loop over the results of something like `find test -type d`.

Reply:
uhuh.

Reply:
I see. If it helps, we could have a "blacklist" of folders that would be ignored in such for loop, similar to what we have in this script: https://github.com/maistra/proxy/blob/89231677979b3d09186ce958bc68685e5e8129c7/maistra/scripts/update-deps.sh#L33

This could be done in a follow up though, as an improvement. The idea is that by not hard coding test/folder names we would be future-proof.